src/server.c

```diff
@@ -946,6 +946,22 @@ struct redisCommand redisCommandTable[] = {
      "read-only @geo",
      0,NULL,1,1,1,0,0,0},
 
+    {"geobbox",geobboxCommand,-7,
+     "write @geo",
+      0,georadiusGetKeys,1,1,1,0,0,0},
+
+    {"geobbox_ro",geobboxroCommand,-7,
```

Comment:
@madolson Thanks for your review.
The `use-memory` flag has been added, and I think `georadius` and `georadiusbymember` are also needed, so I added it by the way.

About command name, I think using `_RO` at the end can be consistent with georadius, which is more user-friendly.  (geo's user)
@itamarhaber WDYT?

Reply:
Yeah, the _RO is what I'm thinking too despite the lack of beauty in it.

Reply:
these commands with an optional STORE argument are causing a lot of pain (it's hard to flag them as write or read commands, and it's hard to extract the key names from the arguments).

i much rather find an opportunity to deprecate the existing ones (and SORT) and create new ones that are similar to SUNION/SUNIONSTORE (i.e. there are specific variants that are always store or non-store).

so while at that subject (of deprecating things), i would like to suggest the following:
* `GEOFIND src_key ...`
* `GEOFINDSTORE dest_key src_key ... [STOREDIST]`
the rest of the arguments for both are:
`longitude latitude width height m|km|ft|mi BOX|ROUND [BYMEMBER member] [WITHCOORD] [WITHDIST] [WITHHASH] [COUNT count] [ASC|DESC]`

meaning:
1. we only have two commands, one that always stores, and one that never stores. instead of the 4 commands we already have, and the 4 we were going to add.
2. the store variant can either store the results, or the dist (same as the original commands had both STORE and STOREDIST arguments), but the key name is in a fixed position and easy to extract from the command arguments.
3. the commands have a BYMEMBER optional argument (instead of the specific command variant for that)
4. both commands take center long/lat and width/height, and the width/height can either define a rectangle, or an ellipse (or a circle if width==height).

@itamarhaber @yangbodong22011 @filipecosta90 @if we can agree on that, we can summon others for additional discussion / approval.

(haven't yet reviewed the code)

Reply:
+1 to deprecation and new command. 

However, shouldn't by member use the member instead of a provided lat/long? I also think it's worth separating out the notion of a circle and doing by radius. The last nit is that I would do geosearch instead, to me find just means find 1. 
```
GEOSEARCH key [frommember member] [fromloc long lat] [byradius radius unit] [bybox width height unit] [WITHCORD] [WITHDIST] [WITHASH] [COUNT count] [ASC|DESC]
```

Reply:
Thanks for your suggestion, I agree with `GEOSEARCH` and `GEOSEARCHSTORE`:

```
GEOSEARCH key [frommember member] [fromloc long lat] [byradius radius unit] [bybox width height unit] [WITHCORD] [WITHDIST] [WITHASH] [COUNT count] [ASC|DESC]

GEOSEARCHSTORE dest_key src_key [frommember member] [fromloc long lat] [byradius radius unit] [bybox width height unit] [WITHCORD] [WITHDIST] [WITHASH] [COUNT count] [ASC|DESC] [STOREDIST]
```
ps
- `frommember` and` fromloc` cannot exist at the same time and one must exist
- `byradius` and `bybox` are the same as above.

Reply:
sounds good to me. so if we go that way we don't have to support ellipse (single radius).
i guess we can add both horizontal and vertical radius, but i don't think that's needed.
also, this syntax allows extending it in the future, like supporting non-axis-aligned bounding box too.

Reply:
feels odd to have this discussion inside a PR review comment. i'll move it to the [issue](https://github.com/redis/redis/issues/4417#issuecomment-737034286) (which has more watchers too):

Reply:
GEOSEARCH/GEOSEARCHSTORE has been added.